ambari-server/src/main/java/org/apache/ambari/server/topology/ConfigRecommendationStrategy.java

```diff
@@ -19,24 +19,51 @@
 package org.apache.ambari.server.topology;
 
 public enum ConfigRecommendationStrategy {
+
   /**
    *  Configuration recommendations are always applied, overriding stack defaults and
    *  configuration defined by the user in the Blueprint and/or Cluster Creation Template.
    */
-  ALWAYS_APPLY,
+  ALWAYS_APPLY(true, true, true),
   /**
    * Configuration recommendations are ignored with this option, both for stack defaults
    * and configuration defined by the user in the Blueprint and/or Cluster Creation Template.
    */
-  NEVER_APPLY,
+  NEVER_APPLY(false, false, false),
   /**
    *  Configuration recommendations are always applied for properties listed as stack defaults,
    *  but not for configurations defined by the user in the Blueprint and/or Cluster Creation Template.
    */
-  ONLY_STACK_DEFAULTS_APPLY,
+  ONLY_STACK_DEFAULTS_APPLY(false, true, false),
   /**
    *  Configuration recommendations are always applied, overriding stack defaults but they don't
    *  override configuration defined by the user in the Blueprint and/or Cluster Creation Template.
    */
-  ALWAYS_APPLY_DONT_OVERRIDE_CUSTOM_VALUES;
+  ALWAYS_APPLY_DONT_OVERRIDE_CUSTOM_VALUES(true, true, false);
+
+  public static ConfigRecommendationStrategy getDefault() {
+    return NEVER_APPLY;
+  }
+
+  private final boolean useStackAdvisor;
+  private final boolean useStackDefaults;
+  private final boolean overrideCustomValues;
+
+  public boolean shouldUseStackAdvisor() {
+    return useStackAdvisor;
+  }
+
+  public boolean shouldUseStackDefaults() {
```

Comment:
I agree with @adoroszlai 's comments.  This config strategy should never provide a way to control if the stack defaults are to be used.  Defaults should always be applied, and this strategy should provide a way to control how the stack advisor's output is applied.